pyblish/util.py

```diff
@@ -73,16 +125,42 @@ def publish(context=None, plugins=None, targets=None):
     }
 
     # Second pass, the remainder
+    plugin_processing = None
+    instances_processed_count = 0
+    total_plugins = len(plugins) + len(collectors)
     for Plugin, instance in logic.Iterator(plugins, context, state):
         try:
             result = plugin.process(Plugin, context, instance)
 
+            # Calculate percentage
+            instances_count = len(logic.instances_by_plugin(context, Plugin))
+
+            instances_processed_fraction = 0
+            if instances_count != 0:
+                instances_processed_fraction = (
+                    float(instances_processed_count) / instances_count
+                )
+
+            plugin_instance_processed = (
+                instances_processed_fraction + plugin_processed_count
+            )
+            percentage = plugin_instance_processed / total_plugins
+
+            # Yield results
+            yield (percentage, result, context)
+
+            # Increment for next iteration
+            instances_processed_count += 1
+            if plugin_processing != Plugin:
+                plugin_processed_count += 1
+                instances_processed_count = 0
+                plugin_processing = Plugin
         except StopIteration:  # End of items
             raise
 
-        except:  # This is unexpected, most likely a bug
+        except Exception as e:  # This is unexpected, most likely a bug
```

Comment:
Can remove the `as e` and just use `throw`, unless you're going to make use of the `e`. `throw` will default to re-throwing the last thrown exception.

Reply:
Done